crates/bevy_ecs/src/query/access.rs

```diff
@@ -532,4 +606,38 @@ mod tests {
 
         assert!(access_a.eq(&expected));
     }
+
+    #[test]
+    fn filtered_access_extend_or() {
```

Comment:
It's hard for me to understand the exact invariants that this test is upholding. I think that this test either needs more comments, to be split apart, or to be cut due to being tightly coupled to the internal API.

Reply:
Added more comments to the test, I hope it makes more sense now as a whole

Reply:
These are much better now, thank you.